add: Изменение кровотечения в игре (часть 1)

vovakons opened the pull request:
## Что этот ПР делает

Кровотечения теперь перешли на части тела, то есть отдельно кровоточат части тела исходя из значения кровотечения на части тела (*раньше было вычисление исходя из брут урона*).
Маленький ребаланс величины кровотечения:
```
Регенерация крови: 0.1 юнит в тик (осталось таким же)
Кровотечение за застрявший предмет: 0.3 юнитов в тик (было 0.5)
Кровотечение за открытую часть тела: 0.75 (было 0.5)
Потеря крови за внутреннее кровотечение части тела: 0.5 (осталось таким же)
Регенерация кровотечения в части тела: 0.01 кровотечения в тик (раньше было 0.1 для всего тела)
//Ниже новые коэффициенты исходя из нового алгоритма
Минимальный урон для получения кровотечения: 10 брут урона
Коэффициет конвертации брут урона на кровотечения: 0.05
Коэффициент конвертации лечения брута при уменьшении кровотечения: 0.01
Шанс получения кровотечения при получени брут урона: 25 + брут_урон * 2.5
```
Изменен предмет бинт:
- использование занимает 3 секунды
- веременно уменьшает величину кровотечения на части тела
- уменьшение кровотечения 1.5, длительность 3 минуты

Простыми словами о кротечениях:
- Части тела получают кровотечение если входящий урон больше 10 с определенным шансом зависящим от величины урона (*больше 30 брут урона единовременно - гарант кровотек)
- Острые предметы наносят кровотечение гарантированно если урона больше 10, также наносят в 2 раза больше кровотечения чем остальные.
- Для временной остановки кровотечения нужно через шифтклик узнать какие части тела кровоточат и наложить на них бинты, пока кровотечение не будет остановлено (*будет сообщение в чате*)
- Сама величина кровотечения медленно уменьшается со временем (0.01 в тик), также уменьшается при лечении брут урона чем угодно.

## Почему это хорошо для игры

Кровотечения сейчас в игре вообще никак не влияют на геймплей, просто ткни в себя бинтом и бегай без кровотечения пока таймер временной остановки кровотека не истечет, кровотечение любой силы устраняется просто тычком одного бинта.

После этого пра нужно будет следить где кровотечение и тратить больше бинтов чтобы остановить кровотечение. Также балансировка коэффициентов дает большее значение кровотечения при больом количестве различных ран, что делает кровотечение действительно опасным.

Слабые удары не наносят кровотечения (*типа резиновых пуль и кулаков*), что вполне логично. Ну и не сильно усложняет механику, слабые кровотечения также легко устраняются бинтами (в одной штуке там 6 пприменений).

*Ну и после этого пра в планах еще парочка по дальнейшей переделке типа артериального кровотека, новых предметов и прочего.*

## Тестирование

На локалке только, наверное лучше через тестмерж в порядке очереди.

vovakons committed:
bleeding refactor

vovakons committed:
bleeding balance

vovakons committed:
bleeding to external body parts as value

vovakons committed:
probe to bleeding for 10-30 brute damage

vovakons committed:
medical banadage and new bleeding

littleboobs commented on code/modules/surgery/organs/blood.dm:
```diff
@@ -16,76 +48,111 @@
+		suppress_amount = amount
+		to_chat(user, span_warning("Повязка уменьшает кровотечение на [declent_ru(PREPOSITIONAL)] у [target], но не может полностью остановить его."))
+	else
+		to_chat(user, span_warning("Повязка временно останавливает кровотечение на [declent_ru(PREPOSITIONAL)] у [target]."))
```
Странные сообщения.

vovakons replied:
> Странные сообщения.

предложи лучше. Нужна какая-то индикация что бинт полностью закрыл кровотек или закрыл его частично.

vovakons committed:
suture and emergency bandage

vovakons committed:
medicine use duration

vovakons committed:
rejuvenate remove bleeding

vovakons committed:
more localization

vovakons committed:
examine bleeding show

vovakons committed:
medical stacks bugfix

littleboobs commented on code/__DEFINES/ru_lang_rules.dm:
```diff
@@ -1,7 +1,7 @@
+#define DATIVE 3 // Дательный: дать полный доступ кому? Клоуну и ассистухе. Идти к чему? Идти к дому.
+#define ACCUSATIVE 4 // Винительный: обвинить кого? Клоуна и ассистуху. Читаю что? Читаю книгу.
+#define INSTRUMENTAL 5 // Творительный: возить по полу кем? Клоуном и ассистухой. Бью чем? Бью монтировкой.
+#define PREPOSITIONAL 6 // Предложный: прохладная история о ком? О клоуне и об ассистухе. Лежать на чём? Лежать на диване.
```
В этом нет никакого смысла, поскольку комментарии для отображения при наведении, должны быть выше объявления и на тройном `///`

littleboobs commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -323,6 +323,23 @@
+			else if(high_bleeding)
+				status += " обильно кровоточ[pluralize_ru(gender, "ит", "ат")]"
+			else
+				status += " кровоточ[pluralize_ru(gender, "ит", "ат")]"
```
бодипарт.гендер?

littleboobs commented on code/modules/mob/living/carbon/human/examine.dm:
```diff
@@ -376,10 +376,19 @@
+		else if (high_bleeding)
+			msg += span_warning("<b>У [genderize_ru(gender, "него", "неё", "него", "них")] обильно кровоточ[pluralize_ru(gender, "ит", "ат")] [bodypart.declent_ru(NOMINATIVE)]!</b>\n")
+		else
+			msg += span_warning("<b>У [genderize_ru(gender, "него", "неё", "него", "них")] кровоточ[pluralize_ru(gender, "ит", "ат")] [bodypart.declent_ru(NOMINATIVE)]!</b>\n")
```
попробуй вместо тегов пользоваться `span_bold`

littleboobs commented on code/modules/surgery/organs/blood.dm:
```diff
@@ -4,88 +4,152 @@
+			if(prob(5))
+				var/symptom = pick("слабость",
+					"лёгкое головокружение",
+					"небольшую тошноту")
```
симптомы одинаковые

littleboobs commented on code/modules/surgery/organs/organ.dm:
```diff
@@ -178,6 +184,8 @@
 		rejuvenate()
 		return ATTACK_CHAIN_PROCEED_SUCCESS
 
+	//TODO остановление кровотечения с помощью бинтов
```
?

vovakons committed:
localization fixes

shahribek commented:
этот ПР мне поможет в будущем

vovakons committed:
Merge branch 'master220' into bleeding-part-1

vovakons committed:
Merge branch 'master220' into bleeding-part-1

vovakons committed:
trauma kits sprites fix

vovakons committed:
medical stacks merge bug fixed

vovakons committed:
ballon alert for bleeding heal

vovakons committed:
advanced suture kit

vovakons committed:
balance

vovakons committed:
balance #2

vovakons committed:
decrease bleeding effect mod

vovakons committed:
increase medicals

vovakons committed:
min brute damage with bleeding

vovakons committed:
normal heal decrease bleeding mod balance

vovakons committed:
improve kits

vovakons committed:
more medicine in vendor

vovakons committed:
burn damage heal bleeding

vovakons committed:
balance and fixes

Anorak2024 commented on code/game/objects/items/stacks/medical.dm:
```diff
@@ -126,114 +136,171 @@
+	max_amount = 4
 	stop_bleeding = 0
+	use_duration = 0
+	use_duration = 2 SECONDS
```
Эээ... Это выглядит неправильно.
Кстати, может сократить?

Anorak2024 commented:
Ещё жаловались что нифига не лечат киты теперь. Но хз, может это они что-то не так сделали.

Anorak2024 commented:
Эээ, и 5+ кровотечения в цикл это как-то овердофига, с учётом того, что оно само восстанавливается без обращения в мед отдел в 50+ раз дольше.

Xetto (Бархатный Рассвет Молодой Звезды) ([следовать]) стонет, "пиздец кровь вытекает"

vovakons committed:
fixes and add borg suture

vovakons commented:
> Эээ, и 5+ кровотечения в цикл это как-то овердофига, с учётом того, что оно само восстанавливается без обращения в мед отдел в 50+ раз дольше.
> 
> Xetto (Бархатный Рассвет Молодой Звезды) ([следовать]) стонет, "пиздец кровь вытекает"

уменьшил

vovakons commented:
Фиксы подвез
<img width="516" height="613" alt="image" src="https://github.com/user-attachments/assets/09d098ef-fd6d-4b42-873b-cf9f00424b54" />

vovakons committed:
rebalance

vovakons committed:
more balance

vovakons committed:
Merge branch 'master220' into bleeding-part-1

vovakons committed:
cargo blood crates for credits

vovakons committed:
robot bodypart bleeding bugfix

vovakons committed:
combat bandage in boxes and vendings

vovakons committed:
bleeding alert on hud

vovakons committed:
medhud bleeding icon

vovakons committed:
bruising use with move

vovakons committed:
suture operation

vovakons committed:
Merge branch 'master220' into bleeding-part-1

vovakons committed:
craft medicals

vovakons committed:
craft cloth from watcher sinew and leather

vovakons committed:
add reagent traneksam acid

vovakons committed:
heal kits craft balance

vovakons committed:
bleeding balance small fix

vovakons committed:
traneksam acid bugfixes

vovakons committed:
Merge branch 'master220' into bleeding-part-1

vovakons committed:
heart stop with overdose traneksam acid

vovakons committed:
blood cargo crates price fix

vovakons committed:
cargo blood crates price balance

vovakons committed:
fix self examine

vovakons committed:
trankecsam acid autoinjector

vovakons committed:
kits craft fix

vovakons committed:
balance

vovakons committed:
medical herbals nerf

vovakons committed:
kits craft fix

vovakons committed:
recipes name fix

vovakons committed:
military emergency bandage bugfix

vovakons committed:
Merge branch 'master220' into bleeding-part-1

vovakons committed:
bleeding balance

Усиление эффекта кровотечения:
- реген крови: с 0.25 на 0.15
- реген кровотека: с 0.025 на 0.01
- модификатор бьерн урона для прижигания: с 0.2 до 0.15

vovakons committed:
remove traneksam autoinjector from crew first aid

Удален инжектор транекзамовой кислоты из аптечки экипажа.

vovakons committed:
transeksam acid effect decrease

Ослаблен эффект лечения кровотечения.
Пока что достаточно 75% понижения кровопотери.

vovakons committed:
bruise mechanic fix

Бинты теперь не останавливают кровотечение полностью.
Теперь бинты ослабляют кровотечение части тела на 80%.

vovakons committed:
Merge branch 'master220' into bleeding-part-1

vovakons committed:
add synth kit

Добавлен новый продвинутый набор из синтплоти, лечит 20 ожогов и 20
брута.

dageavtobusnick committed:
Merge remote-tracking branch 'upstream/master220' into pr/7414
